src/platform/packages/private/kbn-esql-editor/src/esql_editor.tsx

```diff
@@ -302,14 +302,34 @@ const ESQLEditorInternal = function ESQLEditor({
     }
   }, [variablesService, controlsContext, esqlVariables]);
 
-  const showSuggestionsIfEmptyQuery = useCallback(() => {
-    if (editorModel.current?.getValueLength() === 0) {
-      setTimeout(() => {
-        editorRef.current?.trigger(undefined, 'editor.action.triggerSuggest', {});
-      }, 0);
-    }
+  const triggerSuggestions = useCallback(() => {
+    setTimeout(() => {
+      editorRef.current?.trigger(undefined, 'editor.action.triggerSuggest', {});
+    }, 0);
   }, []);
 
+  const maybeTriggerSuggestions = useCallback(() => {
+    const { current: editor } = editorRef;
+    const { current: model } = editorModel;
+
+    if (!editor || !model) {
+      return;
+    }
+
+    const position = editor.getPosition();
+    if (!position) {
+      return;
+    }
+
+    const { lineNumber, column } = position;
+    const lineContent = model.getLineContent(lineNumber);
+    const charBeforeCursor = column > 1 ? lineContent[column - 2] : '';
+
+    if (charBeforeCursor === ' ') {
```

Comment:
there can also be more characters (comma?)

Reply:
Let's leave it as it is for now and we can add more later after playing with it (and receive possible feedback)